File: hangman.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;
int main(){
  cout<<"Welcome to Hangman!"<<endl;
  vector<string> words={"apple","banana","mango","orange"};
  srand(time(0));
  string secret_word=words[rand()%words.size()];
  int tries=6;
  vector<char> guessed_letters;
  while(tries>0){
      string display="";
      for(char i:secret_word){
          if(find(guessed_letters.begin(),guessed_letters.end(),i)!=guessed_letters.end()){
              display+=i;
          }
          else{
              display+=" _";
          }
      }
      cout<<"word :"<<display<<endl;
      // check win
      if(display.find('_')==string::npos){
          cout<<"You guessed the word!"<<endl;
          return 0;
      }
      cout<<"Guess a letter :";
      char letter;
      cin>>letter;
      
      // check if the letter is already existed
      if(find(guessed_letters.begin(),guessed_letters.end(),letter)!=guessed_letters.end()){
          cout<<"letter is already guessed."<<endl;
      }
      guessed_letters.push_back(letter);
      // check if the guessed the wrong letter
      if(secret_word.find(letter)==-1){
          tries--;
          cout<<"wrong guess, tries left :"<<tries<<endl;
      }
  }
  cout<<"You lost, the word was :"<<secret_word<<endl;
}

File: hangman.py
import random
print("Welcome to Hangman!")
words=["apple","banana","mango","orange"]
secret_word=random.choice(words)
tries=6
guessed_letters=[]
while(tries>0):
  display=""
  for i in secret_word:
    if i in guessed_letters:
      display+=i
    else:
      display+=" _"
  print("word :",display)
  # check win
  if "_" not in display:
    print("You guessed the word!!")
    break
  letter=input("Guess a letter :")

  # check if the letter is already existed
  if letter in guessed_letters:
    print("letter is already guessed.")
  guessed_letters.append(letter)

  #check if the guessed the wrong letter
  if letter not in secret_word:
    tries-=1
    print(f"wrong guess, tries left {tries}")
else:
  print("You lost, the word was :",secret_word)